plugins/database/redshift/redshift_test.go

```diff
@@ -331,79 +383,73 @@ func TestPostgresSQL_SetCredentials(t *testing.T) {
 		t.Fatal(err)
 	}
 	dbUser := "vaultstatictest-" + fmt.Sprintf("%s", uid)
-	createTestPGUser(t, url, dbUser, "1Password", testRoleStaticCreate)
+	createTestPGUser(t, connURL, dbUser, "1Password", testRoleStaticCreate)
 
 	db := newRedshift(true)
 	_, err = db.Init(context.Background(), connectionDetails, true)
 	if err != nil {
 		t.Fatalf("err: %s", err)
 	}
 
-	password, err := db.GenerateCredentials(context.Background())
-	if err != nil {
-		t.Fatal(err)
-	}
+	const password1 = "MyTemporaryRootPassword1!"
+	const password2 = "MyTemporaryRootPassword2!"
 
-	usernameConfig := dbplugin.StaticUserConfig{
+	updateReq := newdbplugin.UpdateUserRequest{
 		Username: dbUser,
-		Password: password,
+		Password: &newdbplugin.ChangePassword{
+			NewPassword: password1,
+			Statements:  newdbplugin.Statements{},
+		},
 	}
 
 	// Test with no configured Rotation Statement
-	username, password, err := db.SetCredentials(context.Background(), dbplugin.Statements{}, usernameConfig)
-	if err == nil {
```

Comment:
There is a behaviour change around this, but IMO this check was enforcing buggy behaviour. Both previously and now, the plugin attempted to use a default rotation statement when none is provided, but previously it used a template that had `{{username}}`, and tried to substitute in `{{name}}`. We substitute in both now, because there was inconsistency in the templated variable used between `RotateRootCredentials` and `SetCredentials`, and those got combined into one function.

So the upshot is now users can expect credential rotation to work by default when they don't provide a custom rotation statement, and the change from `err == nil` to `err != nil` here reflects that. WDYT?

Reply:
:+1: